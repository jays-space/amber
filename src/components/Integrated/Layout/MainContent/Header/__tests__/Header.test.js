import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Header } from "..";

describe("<Header />", () => {
  it("should render correctly", () => {
    render(<Header />);

    const headerComponent = screen.getByTitle("header");
    expect(headerComponent).toBeInTheDocument();

    expect(headerComponent).toMatchSnapshot();
  });

  it("should simulate menu toggle", () => {
    render(<Header />);

    const button = screen.getByTestId("button-icon-light");
    fireEvent.click(button);
    expect(button).toBeInTheDocument();
  });
});
