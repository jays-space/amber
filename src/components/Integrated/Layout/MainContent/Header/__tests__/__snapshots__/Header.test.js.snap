// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly 1`] = `
<header
  class="relative flex flex-row justify-between items-center px-5 md:px-10 lg:px-20 py-3 bg-white shadow"
  title="header"
>
  <div
    class="flex flex-1"
  >
    <div
      class="md:hidden"
    >
      <img
        alt="amber software logo"
        class="w-32"
        src="/logo.png"
      />
    </div>
    <div
      class="relative hidden md:flex items-center gap-4"
    >
      <svg
        class="text-slate-500 absolute ml-2 pointer-events-none"
        data-testid="icon-search"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
        />
      </svg>
      <input
        class="w-full py-4 pl-10 pr-20 border-none text-xl focus-visible:outline-slate-400"
        placeholder="search..."
        type="search"
      />
    </div>
  </div>
  <div
    class="ml-10 hidden md:flex flex-row items-center"
  >
    <button
      class="w-10 lg:w-14 aspect-square flex justify-center items-center rounded-full last-of-type:mx-4 cursor-pointer transform ease-out duration-100 cursor-pointer bg-yellow-400 hover:bg-opacity-80"
      data-testid="header-button-primary"
    >
      <svg
        class="text-slate-500 !text-slate-800"
        data-testid="icon-add"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
        />
      </svg>
    </button>
    <button
      class="w-10 lg:w-14 aspect-square flex justify-center items-center rounded-full last-of-type:mx-4 cursor-pointer transform ease-out duration-100 cursor-pointer bg-neutral-200 hover:bg-opacity-80"
      data-testid="header-button-secondary"
    >
      <svg
        class="text-slate-500 "
        data-testid="icon-notification"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z"
        />
      </svg>
    </button>
    <img
      alt="duduzile hlongwane's avatar"
      class="w-14 aspect-square rounded-full"
      src="https://media-exp1.licdn.com/dms/image/C4D03AQELG5uq0_LIug/profile-displayphoto-shrink_100_100/0/1637782454380?e=1667433600&v=beta&t=976dwrMZyY3VL1L48bQwAOvR730al8Ad2-vGvz4pgE8"
    />
  </div>
  <div
    class="z-50 md:hidden"
  >
    <button
      class="flex flex-row justify-center items-center mx-2 last-of-type:mr-0 py-2 px-6 text-slate-800 text-base font-bold tracking-widest rounded-md capitalize cursor-pointer hover:bg-opacity-75 transform ease-out duration-100 w-11 aspect-square !p-2 bg-white hover:bg-slate-50 active:bg-slate-200"
      data-testid="button-icon-light"
    >
      <svg
        class="text-slate-500 text-slate-700"
        data-testid="icon-menu"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
        />
      </svg>
    </button>
  </div>
  <div
    class="fixed md:hidden top-0 translate-x-full bottom-0 px-10 flex flex-col justify-center items-center w-screen h-screen bg-slate-900 bg-opacity-90 backdrop-blur-md z-40 transition-transform ease-out duration-300"
  >
    <div
      class="mb-0"
    >
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          home
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          messages
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          calendar
        </p>
      </div>
    </div>
    <div
      class="mb-0"
    >
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          tasks
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          projects
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          releases
        </p>
      </div>
    </div>
    <div
      class="mb-0"
    >
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          employees
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  bg-yellow-500 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize !text-slate-900 !font-bold !tracking-wider"
          data-testid="paragraph"
        >
          candidates
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          report
        </p>
      </div>
      <div
        class="flex justify-center items-center my-2 py-3 px-6 rounded-md cursor-pointer  hover:bg-slate-800 transform ease-out duration-100"
      >
        <p
          class="tracking-wide  text-slate-700 capitalize text-white"
          data-testid="paragraph"
        >
          settings
        </p>
      </div>
    </div>
  </div>
</header>
`;
